src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -156,6 +156,32 @@ public IActionResult DrainStatus([FromServices] IScriptHostManager scriptHostMan
             }
         }
 
+        [HttpPost]
+        [Route("admin/host/resume")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevelOrInternal)]
+        public async Task<IActionResult> Resume([FromServices] IScriptHostManager scriptHostManager)
+        {
+            _logger.LogDebug($"Received request to resume a draining host - host status: {scriptHostManager?.State.ToString()}");
+
+            if (!Utility.TryGetHostService(scriptHostManager, out IDrainModeManager drainModeManager)
+                || scriptHostManager?.State != ScriptHostState.Running)
+            {
+                _logger.LogWarning("The host is not in a state where we can resume.");
+                return StatusCode(StatusCodes.Status409Conflict);
+            }
+
+            _logger.LogDebug($"Drain mode enabled: {drainModeManager.IsDrainModeEnabled}");
+
+            if (drainModeManager.IsDrainModeEnabled)
+            {
+                _logger.LogDebug("Starting a new host");
+                await scriptHostManager.RestartHostAsync();
+            }
+
+            var status = new ResumeStatus { HostStatus = new HostStatus { State = scriptHostManager.State.ToString() } };
```

Comment:
I don't think you should include the mostly empty HostStatus in the response - just the ScriptHostState

Reply:
Ah I must have misunderstood your comment about using  a `single HostState member`, ScriptHostState definitely makes more sense to me too. Will update